match/lib/match/encrypt.rb

```diff
@@ -52,7 +52,7 @@ def decrypt_repo(path: nil, git_url: nil, manual_password: nil)
              encrypt: false)
         rescue
           UI.error "Couldn't decrypt the repo, please make sure you enter the right password!"
-          UI.user_error!("Invalid password passed via 'MATCH_PASSWORD'") if ENV["MATCH_PASSWORD"]
+          UI.user_error!("Invalid password passed via 'MATCH_PASSWORD'") if FastlaneCore::Env.truthy?("MATCH_PASSWORD")
```

Comment:
Haha, this is an interesting one ... If the user passes a password like `"false"` this will not be called. I think keeping it how it was is fine in that case (or change it to `unless ENV["MATCH_PASSWORD"].nil?` to make it more explicit)